### ISSUE TITLE
Problems updating react-admin

### ISSUE THREAD

**Issue by Merkure** [marmelab/react-admin#6337]
Big question about updating the library ... Every time I update the library I need to completely clear the node_modules ... Because when I update the version and do npm ci the dependencies don't always install correctly and there are a lot of errors ... It would be nice to write in the documentation how to raise the version correctly ...

**Comment by fzaninotto** [comment]
Please follow the issue template. Without a way to reproduce your error, we can't tell if it's on your side or on the react-admin side.

**Comment by Merkure** [comment]
but how should I reproduce it if dependencies are not cached in online editors?

**Comment by fzaninotto** [comment]
If you can't reproduce it, how can we reproduce it?

**Comment by Merkure** [comment]
I meant that in a real project I can easily reproduce this, but cannot do it through the online editor. 
I found a similar issue https://github.com/marmelab/react-admin/issues/2802. But removing package-lock.json every time is a bad idea ...

**Comment by fzaninotto** [comment]
Then can you create a simple project (the simplest possible) reproducing the issue, and post a link to a gist / a repository with that project?

Also, "the dependencies don't always install correctly and there are a lot of errors" doesn't help. 

So for the last time, please follow the issue template.

**Comment by tpinne** [comment]
I encountered a similar error. I had react-admin locked at `3.12.2` and did a `yarn upgrade react-admin`. After that the project didn't compile anymore because the packages `ra-ui-material` and `@material-ui/icons` were missing.

Deleting the `node_modules` and running `yarn install` didn't solve that problem. The packages are still missing.

After reverting my `yarn.lock` to their state before upgrading `react-admin` I again deleted my `node_modules` folder and ran `yarn install`. Everything went back to normal with all packages installed.

I repeated the process `yarn upgrade react-admin`, `rm -rf node_modules` and `yarn install`. The same packages were missing again.

`git reset --hard`, `rm -rf node_modules` and `yarn install` and I'm back to `3.12.2` with all packages installed.

Running `yarn upgrade react-admin`, `rm yarn.lock` and `yarn install` resolves to a successful build with all packages installed, like mentioned in #2802.

Maybe this helps you guys to narrow it down.
